drivers/flash/spi_nor.c

```diff
@@ -554,15 +602,72 @@ static int exit_dpd(const struct device *const dev)
 	return ret;
 }
 
-/* Everything necessary to acquire owning access to the device. */
-static void acquire_device(const struct device *dev)
+#ifdef CONFIG_SPI_NOR_SUSPEND_RESUME
+static int spi_nor_suspend_resume(const struct device *dev)
+{
+	struct spi_nor_data *const driver_data = dev->data;
+	uint8_t opcode;
+	int ret;
+
+	if (driver_data->suspended) {
```

Comment:
Wouldn't it be more intuitive to just parametrize the spi_nor_suspend_resume with desired state?

Reply:
And also pass the state to the driver_data->suspend?